### PR TITLE
Adds Elasticsearch 7.5 to ECE tools for Magento 2.3.5

### PR DESCRIPTION
**Pull request by aepod**
### Description
Adds Elasticsearch 7.5 to ECE tools for Magento 2.3.5

This is related to:  https://github.com/magento/magento-cloud-docker/pull/143

### Fixed Issues (if relevant)
1. magento/magento-cloud-docker#141 Add Elasticsearch 7.5 support
2. magento/ece-tools#681 Add Elasticsearch 7.5 support

### Manual testing scenarios
1.  Test with 2.3.5
2. Set up docker with Elasticsearch 7.5
3. Ensure that the ece-tools does not throw warnings about elasticsearch version

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Service/Validator.php [hunk 1]
```diff
@@ -46,7 +46,8 @@ class Validator
         ServiceInterface::NAME_ELASTICSEARCH => [
             '<2.2.0' => '~1.7.0 || ~2.4.0',
             '>=2.2.0 <2.2.8 || 2.3.0' => '~1.7.0 || ~2.4.0 || ~5.2.0',
-            '>=2.2.8 <2.3.0 || >=2.3.1' => '~1.7.0 || ~2.4.0 || ~5.2.0 || ~6.5.0',
+            '>=2.2.8 <2.3.0 || >=2.3.1 <2.3.5' => '~1.7.0 || ~2.4.0 || ~5.2.0 || ~6.5.0',
+            '>=2.3.5' => '~1.7.0 || ~2.4.0 || ~5.2.0 || ~6.5.0 || ~7.5.0',
```
Comment (YPyltiai) [inline src/Service/Validator.php]: @aepod , I guess this line has redundant `,`

File: config/eol.yaml [hunk 2]
```diff
@@ -24,6 +24,8 @@ elasticsearch:
     eol: 2018-07-31
   - version: '6.5'
     eol: 2020-05-14
+  - version: '7.2'
```
Comment (YPyltiai) [inline config/eol.yaml]: Should be 7.5. Not sure if EOL date changes too with this

**Review by YPyltiai — DISMISSED** [review]

File: config/eol.yaml (hunk 2 above)
Comment (aepod) [reply config/eol.yaml]: Good catch, fixed including date.

**Comment by diazwatson** [comment]
@aepod could you please update this PR with the changes in develop branch?

**Comment by aepod** [comment]
Brought up to date.

**Review by BaDos — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by shiftedreality** [comment]
Approved in https://github.com/magento/magento-cloud-docker/issues/141#issuecomment-584740391

**Pull request merged**